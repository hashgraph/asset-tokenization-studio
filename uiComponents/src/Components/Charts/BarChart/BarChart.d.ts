import { type StackProps as ChakraStackProps } from "@chakra-ui/layout";
import { type ProgressProps as ChakraProgressProps } from "@chakra-ui/progress";
import type { ComponentWithAs } from "@chakra-ui/system";
import { type progressAnatomy as ChakraProgressParts } from "@chakra-ui/anatomy";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
export declare const barChartPartsList: typeof ChakraProgressParts.keys;
type Parts = typeof barChartPartsList;
export interface BarChartThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export interface BarChartProps {
    data: ChakraProgressProps[];
    spacingBars?: ChakraStackProps["spacing"];
    isLoading?: boolean;
    loadingColumnsCount?: number;
}
export declare const BarChart: ComponentWithAs<"div", BarChartProps>;
export {};
