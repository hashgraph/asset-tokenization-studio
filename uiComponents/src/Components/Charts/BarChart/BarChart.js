var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Stack as ChakraStack, } from "@chakra-ui/layout";
import { Progress as ChakraProgress, } from "@chakra-ui/progress";
import { Skeleton as ChakraSkeleton } from "@chakra-ui/react";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
export var barChartPartsList = [
    "label",
    "track",
    "filledTrack",
];
export var BarChart = forwardRef(function (_a, ref) {
    var data = _a.data, _b = _a.spacingBars, spacingBars = _b === void 0 ? "4" : _b, isLoading = _a.isLoading, _c = _a.loadingColumnsCount, loadingColumnsCount = _c === void 0 ? 3 : _c;
    var styles = useChakraMultiStyleConfig("BarChart");
    return (React.createElement(ChakraStack, { ref: ref, sx: styles.container, spacing: spacingBars }, isLoading
        ? Array.from({ length: loadingColumnsCount }).map(function (_, index) { return (React.createElement(ChakraSkeleton, { key: index, width: "".concat(((index + 1) / loadingColumnsCount) * 100, "%"), h: 3 })); })
        : data.map(function (itemProps, index) { return (React.createElement(ChakraProgress, __assign({ key: index }, itemProps, { sx: styles.progress }))); })));
});
