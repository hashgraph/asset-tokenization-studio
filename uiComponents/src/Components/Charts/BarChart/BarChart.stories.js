var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { BarChart } from "./BarChart";
var meta = {
    title: "Design System/Charts/BarChart",
    component: BarChart,
    argTypes: {},
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=775%3A1702"
        },
        docs: {}
    },
    args: {
        max: 100,
        min: 0,
        value: 60,
        isLoading: false
    }
};
export default meta;
var Template = function (args) { return React.createElement(BarChart, __assign({}, args)); };
export var Default = Template.bind({});
Default.args = {
    data: [
        {
            value: 1
        },
        {
            value: 10
        },
        {
            value: 25
        },
        {
            value: 50
        },
        {
            value: 75
        },
        {
            value: 90
        },
        {
            value: 100
        },
    ]
};
