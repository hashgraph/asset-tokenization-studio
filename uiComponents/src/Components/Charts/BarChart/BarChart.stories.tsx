import type { Meta, StoryFn } from "@storybook/react";
import React from "react";

import { BarChart } from "./BarChart";

const meta = {
  title: "Design System/Charts/BarChart",
  component: BarChart,
  argTypes: {},
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=775%3A1702",
    },
    docs: {},
  },
  args: {
    max: 100,
    min: 0,
    value: 60,
    isLoading: false,
  },
} as Meta<typeof BarChart>;
export default meta;

const Template: StoryFn<typeof BarChart> = (args) => <BarChart {...args} />;

export const Default = Template.bind({});

Default.args = {
  data: [
    {
      value: 1,
    },
    {
      value: 10,
    },
    {
      value: 25,
    },
    {
      value: 50,
    },
    {
      value: 75,
    },
    {
      value: 90,
    },
    {
      value: 100,
    },
  ],
};
