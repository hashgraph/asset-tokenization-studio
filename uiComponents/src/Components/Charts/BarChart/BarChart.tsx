import {
  Stack as ChakraStack,
  type StackProps as ChakraStackProps,
} from "@chakra-ui/layout";
import {
  Progress as ChakraProgress,
  type ProgressProps as ChakraProgressProps,
} from "@chakra-ui/progress";
import type { ComponentWithAs } from "@chakra-ui/system";
import { Skeleton as ChakraSkeleton } from "@chakra-ui/react";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";

import { type progressAnatomy as ChakraProgressParts } from "@chakra-ui/anatomy";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";

export const barChartPartsList: typeof ChakraProgressParts.keys = [
  "label",
  "track",
  "filledTrack",
];

type Parts = typeof barChartPartsList;

export interface BarChartThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

export interface BarChartProps {
  data: ChakraProgressProps[];
  spacingBars?: ChakraStackProps["spacing"];
  isLoading?: boolean;
  loadingColumnsCount?: number;
}

export const BarChart: ComponentWithAs<"div", BarChartProps> = forwardRef<
  BarChartProps,
  "div"
>(
  (
    {
      data,
      spacingBars = "4",
      isLoading,
      loadingColumnsCount = 3,
    }: BarChartProps,
    ref
  ) => {
    const styles = useChakraMultiStyleConfig("BarChart");
    return (
      <ChakraStack ref={ref} sx={styles.container} spacing={spacingBars}>
        {isLoading
          ? Array.from({ length: loadingColumnsCount }).map((_, index) => (
              <ChakraSkeleton
                key={index}
                width={`${((index + 1) / loadingColumnsCount) * 100}%`}
                h={3}
              />
            ))
          : data.map((itemProps, index) => (
              <ChakraProgress key={index} {...itemProps} sx={styles.progress} />
            ))}
      </ChakraStack>
    );
  }
);
