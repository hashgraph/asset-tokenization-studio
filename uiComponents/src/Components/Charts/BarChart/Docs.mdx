import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as BarChartStories from "./BarChart.stories";

<Meta of={BarChartStories} />

<Title />

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `BarChart.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    BarChart: ConfigBarChart,
  },
};
```

```tsx dark
const ConfigBarChart: BarChartThemeConfiguration = {
  parts: barChartPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"label" | "track" | "filledTrack"` (`barChartPartsList`)

## Props

<ArgTypes of={BarChartStories} />

## Stories

<Stories title="" />
