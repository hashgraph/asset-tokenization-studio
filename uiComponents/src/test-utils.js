import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
