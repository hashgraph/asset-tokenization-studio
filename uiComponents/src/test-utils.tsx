import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
