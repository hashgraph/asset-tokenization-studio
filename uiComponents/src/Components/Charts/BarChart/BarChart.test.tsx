import { render } from "@/test-utils";
import React from "react";
import { BarChart, type BarChartProps } from "./BarChart";

const data: BarChartProps["data"] = [
  { value: 10 },
  { value: 20 },
  { value: 30 },
  { value: 40 },
  { value: 50 },
  { value: 60 },
  { value: 70 },
  { value: 80 },
];

describe(`<BarChart />`, () => {
  const spacingBars = { spacingBars: "4" };

  const factoryComponent = (props: BarChartProps) =>
    render(<BarChart {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent({ data });
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly when spacingBars is passed", () => {
    const component = factoryComponent({ data, ...spacingBars });
    expect(component.asFragment()).toMatchSnapshot("Using spacingBars");
  });

  test("renders correctly when isLoading is passed", () => {
    const component = factoryComponent({ data, isLoading: true });
    expect(component.asFragment()).toMatchSnapshot("Using isLoading");
  });
});
