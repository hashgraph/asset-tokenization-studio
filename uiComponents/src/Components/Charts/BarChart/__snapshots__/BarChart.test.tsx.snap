// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChart /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-egoftb"
  >
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 10%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 20%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="30"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 30%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="40"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 40%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 50%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 60%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 70%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="80"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 80%;"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<BarChart /> renders correctly when isLoading is passed: Using isLoading 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-egoftb"
  >
    <div
      class="chakra-skeleton css-z6vw7z"
    />
    <div
      class="chakra-skeleton css-rdh281"
    />
    <div
      class="chakra-skeleton css-12xb90b"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<BarChart /> renders correctly when spacingBars is passed: Using spacingBars 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-egoftb"
  >
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 10%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 20%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="30"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 30%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="40"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 40%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 50%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 60%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 70%;"
      />
    </div>
    <div
      class="css-9mbr6u"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="80"
        class="css-1nsw7s9"
        role="progressbar"
        style="width: 80%;"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
